src/Response/CsvResponse.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-diactoros for the canonical source repository
+ * @copyright Copyright (c) 2015-2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-diactoros/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Zend\Diactoros\Response;
+
+use Psr\Http\Message\StreamInterface;
+use Zend\Diactoros\Exception;
+use Zend\Diactoros\Response;
+use Zend\Diactoros\Stream;
+
+use function get_class;
+use function gettype;
+use function is_object;
+use function is_string;
+use function sprintf;
+
+/**
+ * CSV response.
+ *
+ * Allows creating a CSV response by passing a string to the constructor;
+ * by default, sets a status code of 200 and sets the Content-Type header to
+ * text/csv.
+ */
+class CsvResponse extends Response
+{
+    use DownloadResponseTrait;
+    use InjectContentTypeTrait;
+
+    /**
+     * Create a CSV response.
+     *
+     * Produces a CSV response with a Content-Type of text/csv and a default
+     * status of 200.
+     *
+     * @param string|StreamInterface $text String or stream for the message body.
+     * @param int $status Integer status code for the response; 200 by default.
+     * @param string $filename
+     * @param array $headers Array of headers to use at initialization.
+     */
+    public function __construct($text, int $status = 200, string $filename = '', array $headers = [])
+    {
+        if (is_string($filename) && $filename !== '') {

```

Comment:
It IS string, per param declaration, so we don't need to check again here

Reply:
Good point. I forgot to remove that after setting the parameter type.